pmd-core/src/test/java/net/sourceforge/pmd/cli/CoreCliTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.cli;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.RestoreSystemProperties;
+import org.junit.rules.TemporaryFolder;
+
+import net.sourceforge.pmd.PMD;
+
+/**
+ *
+ */
+public class CoreCliTest {
+
+    private static final String DUMMY_RULESET = "net/sourceforge/pmd/cli/FakeRuleset.xml";
+    private static final String STRING_TO_REPLACE = "__should_be_replaced__";
+
+    @Rule
+    public TemporaryFolder tempDir = new TemporaryFolder();
+    @Rule
+    public RestoreSystemProperties restoreSystemProperties = new RestoreSystemProperties();
+    private Path srcDir;
+
+    @Before
+    public void setup() throws IOException {
+        // set current directory to wd
+        Path root = tempRoot();
+        System.setProperty("user.dir", root.toString());
+
+        // create a few files
+        srcDir = Files.createDirectories(root.resolve("src"));
+        writeString(srcDir.resolve("someSource.dummy"), "dummy text");
+    }
+
+
+    @Test
+    public void testPreExistingReportFile() throws IOException {
+        Path reportFile = tempRoot().resolve("out/reportFile.txt");
+        // now we create the file
+        Files.createDirectories(reportFile.getParent());
+        writeString(reportFile, STRING_TO_REPLACE);
+
+        assertTrue("Report file should exist", Files.exists(reportFile));
+
+        runPmdSuccessfully("-d", srcDir, "-R", DUMMY_RULESET, "-r", reportFile);
+
+        assertNotEquals(readString(reportFile), STRING_TO_REPLACE);
+    }
+
+    @Test
+    public void testNonExistentReportFile() {
+        Path reportFile = tempRoot().resolve("out/reportFile.txt");
```

Comment:
I'd use here a different intermediate directory. If this test runs after the other, then we don't test, whether the directory is created... or we need another test case "testNonExistentReportDirAndFile()"